lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -117,6 +119,113 @@ sub list {
     $self->render(json => {JobTemplates => \@templates});
 }
 
+sub validate_yaml {
+    my ($self, $yaml, $validate_schema) = shift;
+
+    # Note: Using the schema filename; slurp'ed text isn't detected as YAML
+    my $schema_yaml = $self->app->home->child('assets', 'job.yaml')->to_string;
```

Comment:
Have you tested whether that file is installed and packaged correctly? Maybe you need to extend the Makefile and the *.spec file (if that directory is not already covered by some glob expression).

Not sure whether it makes a big difference, but it might make sense to store the `$validator` object persistently and reuse it instead of creating it all the time from scratch.

Reply:
And I don't think it belongs in assets at all - I would put it in public, so you can link to it as $SITE_URL/job.yaml

But it really deserves a better name

Reply:
I moved it to `public/schema/JobTemplate.yaml`